### ISSUE TITLE
velero restore describe CLI doesn't display the volume restored by CSI correctly

### ISSUE THREAD

**Issue by blackpiglet** [vmware-tanzu/velero#7849]
**What steps did you take and what happened:**

There are some volumes restored by CSI way, but the `velero restore describe` CLI doesn't display that information correctly.

```
time="2024-05-31T09:08:48Z" level=info msg="Find BackupVolumeInfo for PV pvc-7b09c932-9b56-4015-9b4c-00321a9393ba." logSource="pkg/restore/restore.go:1200" restore=velero/upgrade-test-20240531170845
time="2024-05-31T09:08:48Z" level=info msg="Dynamically re-provisioning persistent volume because it has a CSI VolumeSnapshot or a related snapshot DataUpload." groupResource=persistentvolumes logSource="pkg/restore/restore.go:1221" 
```

```
./velero restore create --from-backup test-01 --namespace-mappings test:restore2

./velero restore describe test-01-20240531181243 --details  [18:13:37]
Name:         test-01-20240531181243
Namespace:    velero
Labels:       <none>
Annotations:  <none>

Phase:                       Completed
Total items to be restored:  15
Items restored:              15

Started:    2024-05-31 18:12:45 +0800 CST
Completed:  2024-05-31 18:13:19 +0800 CST

Warnings:
  Velero:     <none>
  Cluster:  could not restore, VolumeSnapshotContent "snapcontent-91c8c8d5-9bf4-4dd0-af42-baea97f4ee4c" already exists. Warning: the in-cluster version is different than the backed-up version
            could not restore, VolumeSnapshotContent "snapcontent-fa7c54ee-9426-445c-a0d4-6b72645260e5" already exists. Warning: the in-cluster version is different than the backed-up version
  Namespaces:
    restore2:  could not restore, ConfigMap "kube-root-ca.crt" already exists. Warning: the in-cluster version is different than the backed-up version

Backup:  test-01

Namespaces:
  Included:  all namespaces found in the backup
  Excluded:  <none>

Resources:
  Included:        *
  Excluded:        nodes, events, events.events.k8s.io, backups.velero.io, restores.velero.io, resticrepositories.velero.io, csinodes.storage.k8s.io, volumeattachments.storage.k8s.io, backuprepositories.velero.io
  Cluster-scoped:  auto

Namespace mappings:  test=restore2

Label selector:  <none>

Or label selector:  <none>

Restore PVs:  auto

CSI Snapshot Restores: <none included>

Existing Resource Policy:   <none>
ItemOperationTimeout:       4h0m0s

Preserve Service NodePorts:  auto

Uploader config:


HooksAttempted:   0
HooksFailed:      0

Resource List:
  apps/v1/Deployment:
    - restore2/hello-app(created)
  apps/v1/ReplicaSet:
    - restore2/hello-app-76cf99c74b(created)
  snapshot.storage.k8s.io/v1/VolumeSnapshot:
    - restore2/velero-premium-b7mbz(created)
    - restore2/velero-standard-xqbgj(created)
  snapshot.storage.k8s.io/v1/VolumeSnapshotClass:
    - velero(skipped)
  snapshot.storage.k8s.io/v1/VolumeSnapshotContent:
    - snapcontent-91c8c8d5-9bf4-4dd0-af42-baea97f4ee4c(failed)
    - snapcontent-fa7c54ee-9426-445c-a0d4-6b72645260e5(failed)
  v1/ConfigMap:
    - restore2/kube-root-ca.crt(failed)
  v1/Namespace:
    - restore2(created)
  v1/PersistentVolume:
    - pvc-7b09c932-9b56-4015-9b4c-00321a9393ba(skipped)
    - pvc-96315c08-a6f5-4e24-bd79-8cdabc35c4cb(skipped)
  v1/PersistentVolumeClaim:
    - restore2/premium(created)
    - restore2/standard(created)
  v1/Pod:
    - restore2/hello-app-76cf99c74b-mfpfc(created)
  v1/ServiceAccount:
    - restore2/default(skipped)
```

**What did you expect to happen:**

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`): 
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by reasonerjt** [comment]
The root cause is that there's a wrong assumption that at the time the restoreVolumeInfoTracker is populated all the restored PVCs have been bound to PVs, which may not be the case for CSI snapshotters.

In the struct we should use PVC-CSISnapshot mapping instead of PV-CSISnapshot mapping

**Comment by reasonerjt** [comment]
Removing the `target/1.14-rc.2` label as the PR has been merged to release-1.14 branch.

Will close this issue when the cherry-pick PR is merged., which is currently blocked by DCO issue.
